### PR TITLE
Added Dedicated Game Server Sample (experimental)

### PR DESCRIPTION
**Pull request by RikuTheFuffs**
### Description
This PR includes a new experimental sample that showcases how to use official Unity packages and tools to make a multiplayer game.

### Issue Number(s)
N/A

### Contribution checklist
 - [N/A] Tests have been added for the project and/or any internal package
 - [N/A] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [N/A] JIRA ticket ID is in the PR title or at least one commit message
 - [N/A] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=159) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=159) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: jilfranco-unity<br/>:white_check_mark: RikuTheFuffs<br/>:x: Louis-Philippe Lafontaine-Bédard<br/><hr/>**Louis-Philippe Lafontaine-Bédard** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.bitesize?pullRequest=159) it.</sub>

**Comment by jilfranco-unity** [comment]
@RikuTheFuffs you already know about these, but I'll add my few comments here for others' visibility also :)

* We should call out where users can adjust the min/max amount of players in the readme!
* I've also found a broken link [here](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/1bcb86d0b2d4c53b2f3f29d73e4b75dc0086802a/Experimental/DedicatedGameServer/README.md#:~:text=To%20enable%20autoconnect,if%20client%20checkbox%20.) for the [multiplayer-roles.md](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/1bcb86d0b2d4c53b2f3f29d73e4b75dc0086802a/Experimental/DedicatedGameServer/multiplayer-roles.md)
* The [web badges](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/blob/1bcb86d0b2d4c53b2f3f29d73e4b75dc0086802a/Experimental/DedicatedGameServer/README.md#:~:text=Dedicated%20Game%20Server-,The,-Dedicated%20Game%20Server) in the readme for the engine and netcode versions will need to up updated also!

File: .yamato/_triggers.yml
Comment (fernando-cortez) [inline .yamato/_triggers.yml:5]: Makes sense to add specific yml files for the triggers and the tests.

However, inside the previous build.yml file, the old pull request trigger job remains. I'm fine with removing that one out of the build.yml file and keeping this here, but let's also move over the standard pass that was recently added as a dependency to our pull request triggers.

On that note, project-standards.yml seems to have been removed. Let's revisit that merge and get that file back, so that the job would fire, since the job def file isn't actually compiling on Yamato (both build.yml & project-standards.yml). See the branch here: https://unity-ci.cds.internal.unity3d.com/project/1573/branch/sample%2FDGS?nav=jobDefinitions.

Last note about the project tests, could we leverage UPM-CI tools here, since I'm seeing that there's an individual job to build a project for every supported platform? You can notice that the [dependency graph](https://unity-ci.cds.internal.unity3d.com/job/35312372/dependency-graph) on Boss Room's pull request trigger job has one packing job, in which the 3 project tests refer to that artifact.

I'm thinking, every project is packed, and we use the booleans (run_editor_tests & run_playmode_tests) inside the projects to determine whether a project test is run, like you've done here. So we would shrink by a lot the number of jobs run.

File: Experimental/DedicatedGameServer/Assets/Editor/UI.meta
Comment (fernando-cortez) [inline Experimental/DedicatedGameServer/Assets/Editor/UI.meta:1]: I think the whole Editor folder can be removed, no? It's empty for me.

File: Experimental/DedicatedGameServer/Assets/Plugins/ParrelSync/ScriptableObjects/ParrelSyncProjectSettings.asset
Comment (fernando-cortez) [inline Experimental/DedicatedGameServer/Assets/Plugins/ParrelSync/ScriptableObjects/ParrelSyncProjectSettings.asset:1]: Not sure what this is, it's missing a reference to its script, so this could be removed entirely too?

**Review by fernando-cortez — CHANGES_REQUESTED** [review]
Adding requested changes since some of the Yamato jobs from develop are not removed/not compiling.

File: Experimental/DedicatedGameServer/Assets/Editor/UI.meta
Comment (RikuTheFuffs) [reply Experimental/DedicatedGameServer/Assets/Editor/UI.meta]: Nice catch, removed!

File: Experimental/DedicatedGameServer/Assets/Plugins/ParrelSync/ScriptableObjects/ParrelSyncProjectSettings.asset
Comment (RikuTheFuffs) [reply Experimental/DedicatedGameServer/Assets/Plugins/ParrelSync/ScriptableObjects/ParrelSyncProjectSettings.asset]: Looks like some leftover from LP's first iterations with ParrelSync. Removed!

File: .yamato/_triggers.yml
Comment (RikuTheFuffs) [reply .yamato/_triggers.yml:5]: Disclaimer: these changes were made by LP, months ago, so I don't have the full context of "why" they were made.

> On that note, project-standards.yml seems to have been removed. Let's revisit that merge and get that file back, so that the job would fire

I'll open a new branch to start from a point before the merge (reverting it doesn't seem enough, this PR still tries to delete the Anticipation Sample) and open a new PR. If the issues persist there, we'll fix them.

**Comment by RikuTheFuffs** [comment]
Closing as superseeded by: https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/pull/160v

File: .yamato/_triggers.yml
Comment (RikuTheFuffs) [reply .yamato/_triggers.yml:5]: > On that note, project-standards.yml seems to have been removed.

project-standards is not present in `main` either. It's only in `develop`.
It seems that `develop` hasn't been merged into `main` yet.

**Pull request closed without merging**
